library/management/chat/src/main/java/org/quiltmc/qsl/chat/api/ChatSecurityRollbackSupport.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2023 The Quilt Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.chat.api;
+
+/**
+ * Provides rollback support for various classes relating to chat security. This is mostly used in the case of cancelling
+ * signed messages on the client.
+ */
+public interface ChatSecurityRollbackSupport {
```

Comment:
Should this be an injected interface?

Reply:
hm probably

i think i left it out since its only used internally, but since the idea was to allow extending/replacing the signing system it makes sense